apis.md

```diff
@@ -9,3 +9,10 @@ The following OpenTracing APIs are currently available in the RFC stage:
 
 Please refer to the README files in the respective projects for examples of usage.
 
+### Implementation Notes
+
+OpenTracing APIs intend to express a similar experience to the user, while still being idiomatic to the language or framework. The following concepts apply to all variants.
+
+A **Span** is shared mutable state. It accumulates tags and logs, and can also create children. Finishing a span means capturing its state as a SpanRecord.
+
+A **SpanRecord** is an immutable representation of a completed span. It can be reported directly to an OpenTracing system, or encoded into a representation of an alternate system, such as HTrace or Zipkin.
```

Comment:
>
> nit: these details might be better placed in the terminology page, or
> perhaps in a future Architecture page where we can describe the
> relationships between Construction & Recording layers.
>
good point.. I started this in the terminology page, but didn't want to
distract users (or scare them away!) with this sort of nuance.

> Note that SpanRecord is only meaningful for tracers that extend standard
> tracer, and uses of OpenTracing API should never need to know about
> SpanRecord.
>
Good point.. I missed that. Maybe this should be a doc about "standard
tracer"